### PR TITLE
Add MaximumClaims option to Consumer.

### LINKED ISSUE #1: Support a limit on number of claims
By default Marshal will claim as many as it can until it becomes unhealthy.

A user has requested the ability to put an upper bound. This is useful in the situation where you know that your consumers can never operate with more than X partitions/topics. It does put more work on the part of the client to ensure they never underprovision, but this should be fine.

### PR DESCRIPTION
**Pull request by DrTall**
Fixes #1

### CODE REVIEW

File: marshal/consumer.go [hunk 1]
```diff
@@ -256,16 +280,25 @@ func (c *Consumer) GetCurrentLag() int64 {
...
 		if cl.Claimed() {
 			ct++
 		}
 	}
-	return ct
+	return
+}
+
+// IsClaimLimtReached returns the number of claims actively owned by this Consumer.
+func (c *Consumer) IsClaimLimtReached() bool {
```
Comment (zorkian) [inline marshal/consumer.go]: Do you expect clients to want to call this method? I.e., is there any reason to really expose this?

Also, typo: `Limt` -> `Limit`.

File: marshal/consumer_test.go [hunk 2]
```diff
@@ -342,3 +342,33 @@ func (s *ConsumerSuite) TestFastReclaim(c *C) {
 	sort.Strings(msgs)
 	c.Assert(msgs, DeepEquals, []string{"m3", "p1", "p2", "p3"})
 }
+
+func (s *ConsumerSuite) TestMaximumClaims(c *C) {
+	// Ensure aggressive mode claims all partitions in a single call to claim
```
Comment (zorkian) [inline marshal/consumer_test.go]: Comment is bogus

File: marshal/consumer_test.go [hunk 3]
```diff
@@ -342,3 +342,33 @@ func (s *ConsumerSuite) TestFastReclaim(c *C) {
...
+	s.cn.claimPartitions()
+	c.Assert(s.cn.IsClaimLimtReached(), Equals, true)
+	c.Assert(s.cn.GetNumActiveClaims(), Equals, 2)
+	s.cn.claimPartitions()
+	c.Assert(s.cn.IsClaimLimtReached(), Equals, true)
+	c.Assert(s.cn.GetNumActiveClaims(), Equals, 2)
+}
+
+func (s *ConsumerSuite) TestMaximumGreedyClaims(c *C) {
+	// Ensure aggressive mode claims all partitions in a single call to claim
```
Comment (zorkian) [inline marshal/consumer_test.go]: Comment is bogus

**Comment by zorkian** [comment]
I suspect I'd rather have `GetCurrentLoad` call an internal `getNumActiveClaims` method instead of exposing both. I'm not sure why we need to expose the limit methods?

Generally I think that minimizing the number of external methods is better. We can always add more later, but it's really had to remove them without breaking code.

**Comment by DrTall** [comment]
Ah sure, let's not expose the new one. I gave a decent amount of thought to
the issue of whether I can use the existing method, but I didn't give any
thought to whether the API should be expanded

**Comment by DrTall** [comment]
Hmmm so re-running this made the test pass. I've also been running go test --race on my local repo repeatedly and I do get a failure from time to time. :-/

**Comment by DrTall** [comment]
@zorkian PTAL

**Pull request merged**